src/main/kotlin/com/noumenadigital/npl/cli/commands/registry/TestCommand.kt

```diff
@@ -5,37 +5,64 @@ import com.noumenadigital.npl.cli.exception.CommandExecutionException
 import com.noumenadigital.npl.cli.service.ColorWriter
 import com.noumenadigital.npl.cli.service.SourcesManager
 import com.noumenadigital.npl.cli.service.TestHarness
+import com.noumenadigital.npl.testing.coverage.CoverageAnalyzer
+import com.noumenadigital.npl.testing.coverage.LineCoverageAnalyzer
+import com.noumenadigital.npl.testing.coverage.NoCoverageAnalyzer
+import com.noumenadigital.npl.testing.coverage.SonarQubeReporter
 import org.tap4j.util.StatusValues
+import java.io.File
 import java.time.Duration
 
 data class TestCommand(
-    private val targetDir: String = ".",
-    private val testHarness: TestHarness = TestHarness(SourcesManager(targetDir)),
+    private val params: List<String> = emptyList(),
 ) : CommandExecutor {
     companion object {
         const val MIN_PADDING = 25
+        const val COVERAGE_FILE = "coverage.xml"
     }
 
     override val commandName: String = "test"
     override val description: String = "Run the NPL tests"
 
     override val parameters: List<CommandParameter> =
         listOf(
-            PositionalParameter(
-                name = "directory",
+            NamedParameter(
+                name = "--sourceDir",
                 description = "Source directory containing NPL tests to run",
                 defaultValue = ".",
                 isRequired = false,
+                valuePlaceholder = "<directory>",
+            ),
```

Comment:
Not for now, but just realized that we need this ticket: https://noumenadigital.atlassian.net/browse/ST-4684